### PR TITLE
Deprecate types in get field mapping API

### PR DESCRIPTION
**Pull request by mayya-sharipova**
- Add deprecation warning to RestGetFieldMappingAction
- Add two new java HRLC classes GetFieldMappingsRequest and
GetFieldMappingsResponse. These classes use new typeless forms
of a request and response, and differ in that from the server
versions.

Relates to #35190

### CODE REVIEW

**Review by jtibshirani — COMMENTED** [review]
Thanks @mayya-sharipova, this is looking really nice to me! I'll add @hub-cap as a reviewer as he might want to follow along with the HLRC changes.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetFieldMappingsRequest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    private String[] fields = Strings.EMPTY_ARRAY;
+
+    private boolean includeDefaults = false;
+
+    private String[] indices = Strings.EMPTY_ARRAY;
+
+    private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpen();
+
+    public GetFieldMappingsRequest() {}
```
[the remark this answers is not in the document]
Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetFieldMappingsRequest.java]: is nothing actually required here in the request? If not then it makes sense to remove it

File: client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        Map<String, Map<String, FieldMappingMetaData>> mappings = new HashMap<>();
+        int indices = randomInt(10);
+        for(int i = 0; i < indices; i++) {
+            Map<String, FieldMappingMetaData> fieldMappings = new HashMap<>();
+            int fields = randomInt(10);
+            for(int k = 0; k < fields; k++) {
+                final String mapping = randomBoolean() ? "{\"type\":\"string\"}" : "{\"type\":\"keyword\"}";
+                FieldMappingMetaData metaData =
+                    new FieldMappingMetaData("my field", new BytesArray(mapping));
+                fieldMappings.put("field" + k, metaData);
```
[the remark this answers is not in the document]
Comment (hub-cap) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java]: ++ to this

**Review by hub-cap — APPROVED** [review]
That is some hairy parsing logic from the old code :) You might add a note to the old ones that mentions the new client side ones and how changes to the old parsing logic shouldnt differ from the client side ones as well, just so people know if they change it they should also change these... OH I CANT WAIT TILL WE ARE GENERATING THESE :P 

Im LGTM once @jtibshirani 	is LGTM w the PR.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayya-sharipova) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java]: @jtibshirani  Thanks for the comment. I have removed `FieldMappingMetaData.NULL` in the client `GetFieldMappingsResponse` as it is not necessary.  An empty `FieldMappingMetaData` will indicate that field(s) are not found.

**Comment by mayya-sharipova** [comment]
@jtibshirani   and @hub-cap  thanks for the review. I have addressed all your comments.

@jtibshirani  This is ready for another review, whenever you have time.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetFieldMappingsRequest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.common.Strings;
+
+/** Request the mappings of specific fields */
+public class GetFieldMappingsRequest implements Validatable {
+
+    protected boolean local = false;
```
Comment (jtibshirani) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/indices/GetFieldMappingsRequest.java]: Small comment, can this be `private`?

File: client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        Map<String, Map<String, FieldMappingMetaData>> mappings = new HashMap<>();
+        // if mappings is empty, means that fields are not found
+        if (randomBoolean()) {
+            int indices = randomInt(10);
+            for (int i = 0; i < indices; i++) {
+                Map<String, FieldMappingMetaData> fieldMappings = new HashMap<>();
+                int fields = randomInt(10);
+                for (int k = 0; k < fields; k++) {
+                    final String mapping = randomBoolean() ? "{\"type\":\"string\"}" : "{\"type\":\"keyword\"}";
+                    FieldMappingMetaData metaData = new FieldMappingMetaData("my field", new BytesArray(mapping));
```
Comment (jtibshirani) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetFieldMappingsResponseTests.java]: Small comment, "my field" -> `randomAlphaOfLength(8)`.

**Review by jtibshirani — APPROVED** [review]
Looks good to me as well, just left a couple non-critical comments.

**Pull request merged**